src/mpid/ch4/shm/xpmem/xpmem_control.c

```diff
@@ -99,17 +106,17 @@ int MPIDI_XPMEM_ctrl_send_lmt_req_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr)
         /* store CH4 am rreq info */
         MPIDIG_REQUEST(rreq, buffer) = NULL;
         MPIDIG_REQUEST(rreq, datatype) = MPI_BYTE;
-        MPIDIG_REQUEST(rreq, count) = slmt_req_hdr->data_sz;
-        MPIDIG_REQUEST(rreq, rank) = slmt_req_hdr->src_rank;
-        MPIDIG_REQUEST(rreq, tag) = slmt_req_hdr->tag;
-        MPIDIG_REQUEST(rreq, context_id) = slmt_req_hdr->context_id;
+        MPIDIG_REQUEST(rreq, count) = slmt_rts_hdr->data_sz;
+        MPIDIG_REQUEST(rreq, rank) = slmt_rts_hdr->src_rank;
+        MPIDIG_REQUEST(rreq, tag) = slmt_rts_hdr->tag;
+        MPIDIG_REQUEST(rreq, context_id) = slmt_rts_hdr->context_id;
         MPIDI_REQUEST(rreq, is_local) = 1;
 
         /* store XPMEM internal info */
-        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).src_offset = slmt_req_hdr->src_offset;
-        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).data_sz = slmt_req_hdr->data_sz;
-        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).sreq_ptr = slmt_req_hdr->sreq_ptr;
-        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).src_lrank = slmt_req_hdr->src_lrank;
+        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).src_offset = slmt_rts_hdr->src_offset;
+        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).data_sz = slmt_rts_hdr->data_sz;
+        MPIDI_XPMEM_REQUEST(rreq, sreq_ptr) = slmt_rts_hdr->sreq_ptr;
+        MPIDI_XPMEM_REQUEST(rreq, unexp_rreq).src_lrank = slmt_rts_hdr->src_lrank;
```

Comment:
I just want to save space, so I move `sreq_ptr` attribute in `MPIDI_XPMEM_am_unexp_rreq_t` to `MPIDI_XPMEM_am_request_t`. I use `MPIDI_XPMEM_REQUEST(rreq, sreq_ptr)` in `MPIDI_XPMEM_lmt_cts_isend` when receiver replies CTS packet to sender. I can delete this and change `MPIDI_XPMEM_lmt_cts_isend` API a little bit to avoid this combined design.

Reply:
If you think this is a useful optimization, please separate it into a new commit. Please do not combine two things in a single commit!